### PR TITLE
A blog post for reviewing after a tremendous writer's block

### PR DESCRIPTION
**Pull request by riikkanen**

### CODE REVIEW

File: _posts/2020-12-15-why-developer-should-know-owasp.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+Finally, let's think about implementing a simple feedback form to an existing web site. Well, implementing a simple feedback form sounds quite an easy and straightforward task and it might look like we don't need to consider information security at all. However, a simple feedback form can provide a way for a hacker to harm other parts of the system, for example, by using an SQL injection. The form can also tempt spam bots, either filling your database (and site) with numerous false feedbacks or misusing the form to relay email spam messages to others. Furthermore, if we collect the usernames, phone numbers, and hometown, it quickly becomes a person registry, where the data must be protected.
```
Comment (phusu) [inline _posts/2020-12-15-why-developer-should-know-owasp.md:21]: If you want to add something, then maybe some small reference to GDPR related to the person registry.

File: _posts/2020-12-15-why-developer-should-know-owasp.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+![Bobby Tables we call him](/img/why-owasp/exploits_of_a_mom.png)
+
+*Bobby Tables we call him, a classic from [https://xkcd.com/327/](https://xkcd.com/327/)*
+
+In a software project, the possible risks must be examined carefully, and sufficient security level needs to be decided. If the information system is going to be used in a restricted network by a certain number of users, you might not need the same amount of security as in open banking systems. However, it is a decision which needs to be done knowingly. 
+
+# Where the information security comes from?
+
+The information security in a project comes from people. Depending on the project there might be some security specialists but all the people involved in the project need to aware of security issues.
```
Comment (phusu) [inline _posts/2020-12-15-why-developer-should-know-owasp.md:37]: need to **be** aware of

File: _posts/2020-12-15-why-developer-should-know-owasp.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+One security aspect is the practises used in the project. Do people behave securely? When working on a project, you can't tell everyone about it, at least not at a detailed level. You must not keep your passwords on post-its, you should use always good passwords, perhaps utilize a password manager, and enable multi-factor authentication whenever it's possible. Do not push sensitive data, such as database passwords, to version control. Use a VPN instead of unprotected WLANs. Recognize if the risk is bearable before pasting project-specific things into third-party services, like Pastebin, XML validator or some language translator. After all that the other aspect is being aware of different security threats which exist in a digital environment.
```
Comment (phusu) [inline _posts/2020-12-15-why-developer-should-know-owasp.md:39]: Perhaps: "at least not **on** a detailed level."

File: _posts/2020-12-15-why-developer-should-know-owasp.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+![Passwords on post-its](/img/why-owasp/passwords.jpg)
+
+The developers are basically problem solvers. We need to solve a problem which is given to us. Sometimes the problem is easy and simple to solve. However, just solving the problem might not be enough. For example, a testing specialist might find a bug from the implementation in seconds. They have a different point of view and typically try all the nasty corner cases. Does this work with empty input? Or with negative numbers? If the developer is experienced, they have been already taken care of all those corner cases. Also, hackers have a different point of view. They are used to finding vulnerabilities from the software. To be able to secure the software, also the developers must be aware of the threats.
```
Comment (phusu) [inline _posts/2020-12-15-why-developer-should-know-owasp.md:43]: Change to: "If the developers are experienced, they have already"...

**Review by phusu — APPROVED** [review]
Some small corrections and they might be a matter of taste.

File: _posts/2020-12-15-why-developer-should-know-owasp.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+layout: post
+title: Why a developer should know OWASP?
+author: riikkanen
+excerpt: OWASP stands for Open Web Application Security Project. Why should I be interested in OWASP as a developer? If I'm not a hacker or cybersecurity specialist, and we don't even handle any sensitive information in our project, can I just skip the OWASP? Well, the answer is "nope".
+tags: 
+- security
+- developers
+---
+
+OWASP stands for Open Web Application Security Project. Why should I be interested in OWASP as a developer? If I'm not a hacker or cybersecurity specialist, and we don't even handle any sensitive information in our project, can I just skip the OWASP? Well, the answer is "nope".
```
Comment (denzilferreira) [inline _posts/2020-12-15-why-developer-should-know-owasp.md:11]: Maybe add link to OWASP?
https://owasp.org/

File: _posts/2020-12-15-why-developer-should-know-owasp.md (hunk 1 above)
Comment (denzilferreira) [reply _posts/2020-12-15-why-developer-should-know-owasp.md:21]: web site -> website ?

Comment (riikkanen) [reply _posts/2020-12-15-why-developer-should-know-owasp.md]: added GDPR and fixed typo

File: _posts/2020-12-15-why-developer-should-know-owasp.md (hunk 2 above)
Comment (riikkanen) [reply _posts/2020-12-15-why-developer-should-know-owasp.md]: fixed, thanks!

File: _posts/2020-12-15-why-developer-should-know-owasp.md (hunk 3 above)
Comment (riikkanen) [reply _posts/2020-12-15-why-developer-should-know-owasp.md]: fixed

File: _posts/2020-12-15-why-developer-should-know-owasp.md (hunk 5 above)
Comment (riikkanen) [reply _posts/2020-12-15-why-developer-should-know-owasp.md]: added

File: _posts/2020-12-15-why-developer-should-know-owasp.md (hunk 4 above)
Comment (riikkanen) [reply _posts/2020-12-15-why-developer-should-know-owasp.md]: good point

**Review by denzilferreira — APPROVED** [review]
Looks good!

**Pull request merged**